sklearn/gaussian_process/kernels.py

```diff
@@ -1096,22 +1146,21 @@ def __call__(self, X, Y=None, eval_gradient=False):
             hyperparameter of the kernel. Only returned when eval_gradient
             is True.
         """
-        X = np.atleast_2d(X)
         if Y is not None and eval_gradient:
             raise ValueError("Gradient can only be evaluated when Y is None.")
 
         if Y is None:
-            K = self.noise_level * np.eye(X.shape[0])
+            K = self.noise_level * np.eye(len(X))
```

Comment:
Do we support sparse matrices for X in gps? If so, I think len (X) may be undefined. We have a helper _num_samples for this reason.

Reply:
Fixed.

Reply:
(well, ideally this would be tested and the code would have failed before using `_num_samples`)

Reply:
Actually I couldn't find a place in the documentation that states the support for sparse feature vectors...